TempState.java

```diff
@@ -0,0 +1,143 @@
+import java.util.Arrays;
+
+public class TempState {
```

Comment:
is there no way to just extend the original State class so you don't have to copy over every method?